Hw4.cpp

IT-kakaya opened the pull request.

IT-kakaya committed:
Create ex2.cpp

IT-kakaya committed:
Create ex1.cpp

IT-kakaya committed:
Update ex1.cpp

IT-kakaya committed:
Update ex2.cpp

dmitrii-kruglov commented on ex1.cpp:
```diff
@@ -0,0 +1,38 @@
+        cin >> array[i];
+    }
+    
+    int min = array[0]; //Предроложим, что наименьшее число - первое 
```
BP: достаточно запомнить номер этого элемента

dmitrii-kruglov replied:
А этот остался..

dmitrii-kruglov commented on ex2.cpp:
```diff
@@ -0,0 +1,30 @@
+    cout << "Сумма всех положительных элементов массива:" << endl;
+    cout << sum;
+}
+//отправляю еще раз, чтобы Вы заметили, в коде нет изменений
```
Достаточно было просто меня добавить в rewivers или отметить в комментах. Ещё один коммит, это бессмысленно.

dmitrii-kruglov commented on ex1.cpp:
```diff
@@ -0,0 +1,38 @@
+    
+    int min = array[0]; //Предроложим, что наименьшее число - первое 
+    int Nmin = 0;
+    int max = array[0]; //Предположим, что наибольшее число - второе
```
Так вы предположили в данном случае что наибольшее тоже первое?

IT-kakaya replied:
Мне нужно было сравнивать каждый элемент массива с каким-то числом, удобнее всего было взять первое, понимая, что в любом случае оно не может быть на выходе одновременно и минимальным, и максимальным

dmitrii-kruglov replied:
да, я понимаю, но в  комментарии написали о "втором"

dmitrii-kruglov reviewed (approved):
Решение принято, с небольшими комментариями

IT-kakaya committed:
Update ex1.cpp

Поменяла в комментарии "второе" на "первое", потому что не заметила, что случайно описалась, комментируя. Сохранять только номер ячейки не получится, так как нам нужно сравнивать между собой именно значения, чтобы определить какое наибольшее, а какое наименьшее. Именно для этого я запоминаю и само число, чтобы понять меньше или больше которого, ищет программа.

dmitrii-kruglov reviewed (approved):
Вторая полностью, а вот по первой остался один коммент

IT-kakaya committed:
Update ex1.cpp

Убрала запоминание значений в массиве для сравнивания
